src/components/Pagination/index.tsx

```diff
@@ -0,0 +1,41 @@
+import React, { useEffect, useState } from 'react';
+import { ChevronLeft, ChevronRight } from 'react-feather';
+import { Box } from '../Box';
+import { Paginate } from './styleds';
+
+interface Props {
+  itemsPerPage: number;
+  items: any[];
+  onChange: (items: any[]) => void;
+}
+
+export default function Pagination({ itemsPerPage, items, onChange }: Props) {
+  const [pageCount, setPageCount] = useState(0);
+  const [itemOffset, setItemOffset] = useState(0);
+
+  useEffect(() => {
+    const endOffset = itemOffset + itemsPerPage;
+    onChange(items.slice(itemOffset, endOffset));
+    setPageCount(Math.ceil(items.length / itemsPerPage));
+  }, [itemOffset, itemsPerPage]);
```

Comment:
`Pagination` component doesn't need to know about `items`. Its not responsibility of Pagination component to do `items.slice` in `useEffect`. In `onChange` we should just pass 2 things 
1. activePageNumber
2. itemOffset

Because for e.g. if I want to call some api on page change then i dont have items to pass to Pagination component. I just need to know page number which user selected and that page number we can pass to api to get new items.

In your situation, wherever you are using `<Pagination />` there you need to do `slice` and set it into state to display it.